sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosVectorIndexSpec.java

```diff
@@ -66,6 +74,127 @@ public CosmosVectorIndexSpec setType(String type) {
         checkNotNull(type, "cosmosVectorIndexType cannot be null");
         this.type = type;
         this.jsonSerializable.set(Constants.Properties.VECTOR_INDEX_TYPE, this.type, CosmosItemSerializer.DEFAULT_SERIALIZER);
+
+        // Reset restricted properties if the type is not DiskANN or QuantizedFlat
+        if (!type.equals(CosmosVectorIndexType.DISK_ANN.toString()) && !type.equals(CosmosVectorIndexType.QUANTIZED_FLAT.toString())) {
+            this.quantizationByteSize = null;
+            this.vectorIndexShardKey = null;
+        }
+        if (!type.equals(CosmosVectorIndexType.DISK_ANN.toString())) {
+            this.indexingSearchListSize = null;
+        }
+        return this;
+    }
+
+    /**
+     * Gets the quantization byte size
+     *
+     * @return quantizationByteSize the number of bytes used in product quantization of the vectors.
+     * A larger value may result in better recall for vector searches at the expense of latency.
+     * This applies to index types DiskANN and quantizedFlat. The allowed range for this parameter
+     * is between 1 and 3.
+     */
+    public Integer getQuantizationByteSize() {
+        if (this.quantizationByteSize == null && validateIndexType(false)) {
+            this.quantizationByteSize = this.jsonSerializable.getInt(Constants.Properties.VECTOR_QUANTIZATION_BYTE_SIZE);
+
+            if (this.quantizationByteSize <= 0) {
+                this.quantizationByteSize = null;
+            }
+        }
+        return this.quantizationByteSize;
+    }
+
+    private Boolean validateIndexType(boolean isIndexingSearchListSize) {
+        String vectorIndexType = this.jsonSerializable.getString(Constants.Properties.VECTOR_INDEX_TYPE);
+        if (!isIndexingSearchListSize) {
+            return vectorIndexType.equals(CosmosVectorIndexType.QUANTIZED_FLAT.toString()) ||
+                vectorIndexType.equals(CosmosVectorIndexType.DISK_ANN.toString());
+        }
+        return vectorIndexType.equals(CosmosVectorIndexType.DISK_ANN.toString());
+    }
+
+    /**
+     * Sets the quantization byte size
+     *
+     * @param quantizationByteSize the number of bytes used in product quantization of the vectors. A larger value may
+     *                             result in better recall for vector searches at the expense of latency. This applies
+     *                             to index types DiskANN and quantizedFlat. The allowed range for this parameter is
+     *                             between 1 and 3.
+     * @return CosmosVectorIndexSpec
+     */
+    public CosmosVectorIndexSpec setQuantizationByteSize(Integer quantizationByteSize) {
+        if (validateIndexType(false) && quantizationByteSize != null) {
```

Comment:
Do you need to think about a way to set it back to null if it's already been set?

Reply:
One more aspect: Silently dropping when already set